flink-core/src/main/java/org/apache/flink/configuration/ClusterOptions.java

```diff
@@ -88,21 +93,28 @@
                                     .build());
 
     @Documentation.Section(Documentation.Sections.EXPERT_CLUSTER)
-    public static final ConfigOption<Boolean> HALT_ON_FATAL_ERROR =
-            key("cluster.processes.halt-on-fatal-error")
+    public static final ConfigOption<Boolean> HALT_ON_SYSTEM_EXIT =
+            key("cluster.processes.halt-on-system-exit")
```

Comment:
>Renaming an existing parameter has a bigger impact on users as they possibly have to change their existing configuration files when upgrading.

This was Robert's suggestion and I also had the same concern. But I incorporated that suggestion because this flag seems to be more of advanced expert feature, which may be used by a few individuals, and I see sometimes Flink config names have been changed slightly or largely. However, as we do not know if this feature is used much, I am fine to keep the name same. I am willing to follow votes.

> would be ok with that change if we decide to merge both parameters into one as they are more or less just defining some special exit behavior. What do you think?

Like your following comment, user system exit vs. global system exit behaviors need to be separate for clarity. Those can be under the same `ClusterOptions` but I am still leaning toward having separate configurations.

Having said that, the only alternative we can think is just to keep `cluster.process.halt-on-fatal-error` as is. What do you think @rmetzger ?

Reply:
The reason why I proposed to change the configuration key away from "halt-on-fatal-error" is that the name is quite misleading. It sounds like fatal errors are causing a System.getRuntime().halt(), however, it means "all system exit calls are replaced by halt()".
The behavior in case of a fatal error is described in FatalExitExceptionHandler (I guess).

You are right that changing a configuration variable is not advised.  It seems that there was some discussion on the configuration parameter, when it was introduced (last comments: https://issues.apache.org/jira/browse/FLINK-16510). Let's go with @hwanju's proposal and keep the configuration name as-is.

Reply:
I'm sorry for the back and forth on this!

Reply:
No problem at all. Naming is hard and it's very simple change :)